src/CppUTest/MemoryLeakDetector.cpp

```diff
@@ -513,21 +519,25 @@ char* MemoryLeakDetector::reallocateMemoryAndLeakInformation(TestMemoryAllocator
 
 void MemoryLeakDetector::invalidateMemory(char* memory)
 {
+#ifndef CPPUTEST_DISABLE_HEAP_POISON
```

Comment:
Why is this @ifdef needed?

Reply:
Address Sanitizer, Boundschecker and Appverifier poison all buffers with a pattern, so this is just unnecessary. I have made it a different conditional as some dynamic code analysis tools might not provide this feature (e.g. Valgrind).